auto_inertial_params/proposal.md

```diff
@@ -0,0 +1,414 @@
+
+  //////////////////////////////////////////////////
+  std::optional<gz::math::Inertiald> Mesh::CalculateInertial(double _density,
+      const sdf::ElementPtr _autoInertiaParams, const ParserConfig &_config,
+      sdf::Errors &_errors)
+  {
+    if (this->dataPtr->filePath.empty())
+    {
+      _errors.push_back({
+        sdf::ErrorCode::WARNING,
+        "File Path for the mesh was empty. Could not calculate inertia"});
+      return std::nullopt;
+    }
+
+    const auto &customCalculator = _config.CustomInertiaCalc();
+
+    sdf::CustomInertiaCalcProperties calcInterface = CustomInertiaCalcProperties(
+      _density, *this, _autoInertiaParams);
+
+    return customCalculator(_errors, calcInterface);
+  }
+```
+
+### Configuring the `CalculateInertial()` function behavior
+
+`CalculateInertial()` functions are also added to the `sdf::Root`, `sdf::World`, 
+`sdf::Model`, `sdf::Link`, `sdf::Collision` and `sdf::Geometry` classes. Calling the 
+`CalculateInertial()` function of any class, in turn recursively calls the `CalculateInertial()` 
+for each element below it down the chain. For eg, generally a user might call the `Root::CalculateInertial()` 
+which further calls `World::CalculateInertials()` for all the worlds which in turn 
+calls the `Model::CaluclateInertials()` for all the models in the world and so on until 
+the final `CalculateInertial()` for the respective geometry type is called where the chain ends.
+
+The `sdf::ParserConfig` object is sent down the `CalculateInertial()` chain as a function 
+parameter and is used to get the configuration information like the registered Custom 
+Inertia Calculator. An `enum class` would be added to the `sdf::ParserConfig` with values 
+that would allow the user to configure the behavior of the `CalculateInertial()` function:
+
+```C++
+/// \enum ConfigureCalculateInertial
+/// \brief Configuration options of how CalculateInertial() function
+/// would be used
+enum class ConfigureCalculateInertial
+{
+  /// \brief If this value is used, CalculateInertial() won't be 
+  /// called from inside the Root::Load() function
+  SKIP_CALCULATION_IN_LOAD,
+
+  /// \brief If this values is used, CalculateInertial() would be 
+  /// called and the computed inertial values would be saved 
+  SAVE_CALCULATION
+};
+```
+
+Setting values from the above enum for the `sdf::ParserConfig` object, the user can 
+configure the `CalculateInertial()` functions. For eg: if the configuration is set to 
+`SKIP_CALCULATION_IN_LOAD` (which would be the default configuration), then the 
+`Root::CalculateInertial()` won't be called from within the `Root::Load()`. The user 
+would need to call the `Root:CalculateInertial()` separately after the load is complete. 
+This is also recommended as the inertia calculation uses the `PoseGraph` to resolve the 
+inertial poses for different collisions if they are not in the link frame. 
```

Comment:
I wouldn't say it's recommended because the way we have it set up now, `Root::CalculateInertial` is called in `Root::Load` after the PoseGraph has been built. I would instead say these configurations would allow the user to skip inertial calculation if they know the the DOM object doesn't have a valid PoseGraph.

Reply:
Updated the doc in [52b582e](https://github.com/gazebosim/sdf_tutorials/pull/92/commits/52b582ee7c155e32a76fe28b278d6af3276b5967)